File: src/Cayita.Demo.NavList/DemoNavlist.cs
using System;
using System.Runtime.CompilerServices;
using jQueryApi;

namespace Cayita.Demo
{
	[IgnoreNamespace]
	public class DemoNavlist
	{
		public DemoNavlist ()
		{
		}

		public static void Execute(Atom parent)
		{

			var nb =new NavList ();

			nb.Add ("Tables");
			nb.Add ("Forms");
			nb.Add ("Modals");
			nb.Add ("Panels");
			nb.AddDivider ();
			nb.Add ("Exit");


			var log = new Div ();
			var code = new Div ();

			new Div (d=>{
				d.ClassName="bs-docs-example";
				d.JQuery.Append(nb).Append(log).Append("C# code".Header(4)).Append(code);
				parent.Append(d);
			});

			nb.Selected += (e) => {
				var i = e.CurrentTarget.As<NavItem> ();
				log.Text=  "{0} Clicked".Fmt(i.Text); 
			};

			var rq =jQuery.GetData<string> ("code/demonavlist.html");
			rq.Done (s=> code.Text=s);

		}
	}
}

File: src/Cayita.Demo/modules/DemoNavlist.js
(function() {
	'use strict';
	////////////////////////////////////////////////////////////////////////////////
	// Cayita.Demo.DemoNavlist
	var $DemoNavlist = function() {
	};
	$DemoNavlist.__typeName = 'DemoNavlist';
	$DemoNavlist.execute = function(parent) {
		var nb = Cayita.UI.NavList(null);
		nb.nav.addValue('Tables', null, null, false, null);
		nb.nav.addValue('Forms', null, null, false, null);
		nb.nav.addValue('Modals', null, null, false, null);
		nb.nav.addValue('Panels', null, null, false, null);
		nb.nav.addDivider('divider');
		nb.nav.addValue('Exit', null, null, false, null);
		var log = Cayita.UI.Atom('div');
		var code = Cayita.UI.Atom('div');
		Cayita.UI.Atom('div', null, null, null, function(d) {
			d.className = 'bs-docs-example';
			$(d).append(nb).append(log).append(Cayita.Fn.header('C# code', 4)).append(code);
			parent.append(d);
		});
		nb.nav.add_selected(function(e) {
			var i = e.currentTarget;
			log.set_text(Cayita.Fn.fmt('{0} Clicked', [i.get_text()]));
		});
		var rq = $.get('code/demonavlist.html');
		rq.done(function(s) {
			code.set_text(s);
		});
	};
	global.DemoNavlist = $DemoNavlist;
	ss.initClass($DemoNavlist, {});
})();
